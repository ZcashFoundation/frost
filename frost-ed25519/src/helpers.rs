use crate::Ed25519Sha512;
use ed25519_dalek::Verifier;

#[cfg(test)]
pub fn verify_signature(
    msg: &[u8],
    group_signature: frost_core::Signature<Ed25519Sha512>,
    group_pubkey: frost_core::VerifyingKey<Ed25519Sha512>,
) {
    let sig = {
        let bytes: [u8; 64] = group_signature.to_bytes();
        ed25519_dalek::Signature::from(bytes)
    };
    let pub_key = {
        let bytes = group_pubkey.to_bytes();
        ed25519_dalek::PublicKey::from_bytes(&bytes).unwrap()
    };
    // Check that signature validation has the expected result.
    assert!(pub_key.verify(msg, &sig).is_ok());
}
