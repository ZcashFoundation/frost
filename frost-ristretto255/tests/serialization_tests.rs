#![cfg(feature = "serialization")]

mod helpers;

use frost_ristretto255::{
    keys::{
        dkg::{round1, round2},
        KeyPackage, PublicKeyPackage, SecretShare,
    },
    round1::{SigningCommitments, SigningNonces},
    round2::SignatureShare,
    SigningPackage,
};

use helpers::samples;
use insta::assert_snapshot;

#[test]
fn check_signing_nonces_postcard_serialization() {
    let nonces = samples::signing_nonces();
    let bytes: Vec<_> = nonces.serialize().unwrap();
    assert_snapshot!(hex::encode(&bytes));
    assert_eq!(nonces, SigningNonces::deserialize(&bytes).unwrap());
}

#[test]
fn check_signing_commitments_postcard_serialization() {
    let commitments = samples::signing_commitments();
    let bytes: Vec<_> = commitments.serialize().unwrap();
    assert_snapshot!(hex::encode(&bytes));
    assert_eq!(
        commitments,
        SigningCommitments::deserialize(&bytes).unwrap()
    );
}

#[test]
fn check_signing_package_postcard_serialization() {
    let signing_package = samples::signing_package();
    let bytes: Vec<_> = signing_package.serialize().unwrap();
    assert_snapshot!(hex::encode(&bytes));
    assert_eq!(
        signing_package,
        SigningPackage::deserialize(&bytes).unwrap()
    );
}

#[test]
fn check_signature_share_postcard_serialization() {
    let signature_share = samples::signature_share();
    let bytes = signature_share.serialize();
    assert_snapshot!(hex::encode(bytes));
    assert_eq!(signature_share, SignatureShare::deserialize(bytes).unwrap());
}
#[test]
fn check_secret_share_postcard_serialization() {
    let secret_share = samples::secret_share();
    let bytes: Vec<_> = secret_share.serialize().unwrap();
    assert_snapshot!(hex::encode(&bytes));
    assert_eq!(secret_share, SecretShare::deserialize(&bytes).unwrap());
}

#[test]
fn check_key_package_postcard_serialization() {
    let key_package = samples::key_package();
    let bytes: Vec<_> = key_package.serialize().unwrap();
    assert_snapshot!(hex::encode(&bytes));
    assert_eq!(key_package, KeyPackage::deserialize(&bytes).unwrap());
}

#[test]
fn check_public_key_package_postcard_serialization() {
    let public_key_package = samples::public_key_package();
    let bytes: Vec<_> = public_key_package.serialize().unwrap();
    assert_snapshot!(hex::encode(&bytes));
    assert_eq!(
        public_key_package,
        PublicKeyPackage::deserialize(&bytes).unwrap()
    );
}

#[test]
fn check_round1_package_postcard_serialization() {
    let round1_package = samples::round1_package();
    let bytes: Vec<_> = round1_package.serialize().unwrap();
    assert_snapshot!(hex::encode(&bytes));
    assert_eq!(
        round1_package,
        round1::Package::deserialize(&bytes).unwrap()
    );
}

#[test]
fn check_round2_package_postcard_serialization() {
    let round2_package = samples::round2_package();
    let bytes: Vec<_> = round2_package.serialize().unwrap();
    assert_snapshot!(hex::encode(&bytes));
    assert_eq!(
        round2_package,
        round2::Package::deserialize(&bytes).unwrap()
    );
}
