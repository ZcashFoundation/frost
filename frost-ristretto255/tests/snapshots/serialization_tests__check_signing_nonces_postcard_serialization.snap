---
source: frost-ristretto255/tests/serialization_tests.rs
expression: "hex::encode(&bytes)"
---
00d76ecff5498d4e9311420c903913a56c94a694b8aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa0a498d4e9311420c903913a56c94a694b8aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa0a00d76ecff5e8f69f2ee87ef7c1e54ecf0c08883e39406c0d3fc01eda94116452870e0e6e3be8f69f2ee87ef7c1e54ecf0c08883e39406c0d3fc01eda94116452870e0e6e3b
